internal/generator/image.go

```diff
@@ -107,26 +111,28 @@ func compressImage(collage *image.Image, quality int) (image.Image, error) {
 	return jpeg.Decode(bytes.NewReader(buf.Bytes()))
 }
 
-func CreateCollage[T Drawable](ctx context.Context, albums []T, displayOptions DisplayOptions) (image.Image, error) {
+func CreateCollage[T Drawable](ctx context.Context, collageElements []T, displayOptions DisplayOptions) (image.Image, error) {
 
 	collageWidth := displayOptions.ImageDimension * displayOptions.Columns
 	collageHeight := displayOptions.ImageDimension * displayOptions.Rows
 	dc := gg.NewContext(collageWidth, collageHeight)
 	dc.SetRGB(0, 0, 0)
 	dc.LoadFontFace(fontFile, displayOptions.FontSize)
 
-	for i, album := range albums {
+	for i, collageElement := range collageElements {
 		x := (i % displayOptions.Columns) * displayOptions.ImageDimension
 		y := (i / displayOptions.Columns) * displayOptions.ImageDimension
-		if *album.GetImage() != nil {
-			dc.DrawImage(*album.GetImage(), x, y)
+		img := collageElement.GetImage()
+		if *img != nil {
+			img = resizeImage(ctx, img, uint(displayOptions.ImageDimension), uint(displayOptions.ImageDimension))
```

Comment:
Hmm we probably don't need to resize every image, i.e., we can only resize if it's not the nxm size needed

Reply:
Yeah it doesn't, it'll only resize if it's the incorrect size. In the `resizeImage` function it checks to see if it is already the correct size, and if it is, it just returns.

Reply:
Ah awesome 👍